src/gwc/src/test/java/org/geoserver/gwc/GWCDataSecurityTest.java

```diff
@@ -149,6 +163,76 @@ public void testNoMosaic() throws Exception {
     }
     
     @Test
+    public void testPermissionMosaicTileWmts() throws Exception {
+        GWC.get().getConfig().setSecurityEnabled(true);
+        
+        //first to cache
+        setRequestAuth("cite", "cite");
+        String path = "gwc/service/wmts?LAYER=sf:mosaic&FORMAT=image/png&SERVICE=WMTS&VERSION=1.0.0" +
+        "&REQUEST=GetTile&TILEMATRIXSET=EPSG:4326&TILEMATRIX=EPSG:4326:0&TILECOL=0&TILEROW=0";
+        MockHttpServletResponse response = getAsServletResponse(path);
+        assertEquals("image/png", response.getContentType() );
+        
+        // try again, now should be cached
+        response = getAsServletResponse(path);
+        assertEquals("image/png", response.getContentType());
+
+        // permission must be denied to cite user
```

Comment:
Shouldn't this check that permission is denied after getting the tile cached?  You've cached it for sf:mosaic, but not for sf:mosaic2 .  If nothing's cached already we should be safe as another deeper error will occur.  Your test will still notice the difference between the errors, but I'd feel better if the dangerous case were the one that's tested.